### PR TITLE
docs: Mongo atlas

### PR DESCRIPTION
**Pull request by vblagoje**
- fixes https://github.com/deepset-ai/haystack/issues/7142

Note to reviewer:
Please be super careful and rerun the examples I added.

### CODE REVIEW

File: integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py [hunk 1]
```diff
@@ -13,7 +13,29 @@ class MongoDBAtlasEmbeddingRetriever:
...
 
-    Needs to be connected to the MongoDBAtlasDocumentStore.
+    The similarity is dependent on the vector_search_index used in the MongoDBAtlasDocumentStore and the chosen metric
+    during the creation of the index (i.e. cosine, dot product, or euclidean). See MongoDBAtlasDocumentStore for more
+    information.
+
+    Usage example:
+    ```python
+    import numpy as np
+    from haystack import Document
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py:23]: unused import
```suggestion
```

File: integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py [hunk 2]
```diff
@@ -13,7 +13,29 @@ class MongoDBAtlasEmbeddingRetriever:
...
+    results = retriever.run(query_embedding=np.random.random(768).tolist())
+    print(results["documents"])
+    ```
+
+    The example above retrieves the 10 most similar documents to a random query embedding from the
+    MongoDBAtlasDocumentStore. Note that dimensions of the query_embedding must match the dimensions of the embeddings
+    stored in the MongoDBAtlasDocumentStore.
     """
 
     def __init__(
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py:41]: can we please also document the `ValueError` in the constructor?

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py [hunk 3]
```diff
@@ -19,6 +19,39 @@
 
 
 class MongoDBAtlasDocumentStore:
+    """
+    MongoDBAtlasDocumentStore is a DocumentStore implementation that uses MongoDB Atlas - a fully managed cloud database
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:23]: no need to make advertisement, no? 😁 
In any case I'd add a link so users can explore on their own
```suggestion
    MongoDBAtlasDocumentStore is a DocumentStore implementation that uses [MongoDB Atlas](https://www.mongodb.com/atlas/database).
```

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py [hunk 4]
```diff
@@ -19,6 +19,39 @@
...
+    from haystack_integrations.document_stores.mongodb_atlas import MongoDBAtlasDocumentStore
+
+    store = MongoDBAtlasDocumentStore(database_name="your_existing_db",
+                                      collection_name="your_existing_collection",
+                                      vector_search_index="your_existing_index")
+    print(store.count_documents())
+    ```
+    """
+
     def __init__(
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:55]: can we please document the `ValueError` which is raised here?

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py [hunk 5]
```diff
@@ -108,13 +150,13 @@ def filter_documents(self, filters: Optional[Dict[str, Any]] = None) -> List[Doc
 
     def write_documents(self, documents: List[Document], policy: DuplicatePolicy = DuplicatePolicy.NONE) -> int:
         """
-        Writes documents into to PgvectorDocumentStore.
+        Writes documents into the MongoDB Atlas collection.
 
         :param documents: A list of Documents to write to the document store.
         :param policy: The duplicate policy to use when writing documents.
         :raises DuplicateDocumentError: If a document with the same id already exists in the document store
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:157]: can we please also document the `ValueError`?

Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:157]:
```suggestion
        :raises DuplicateDocumentError: If a document with the same ID already exists in the document store
```

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py [hunk 6]
```diff
@@ -168,6 +210,7 @@ def embedding_retrieval(
         :param query_embedding: Embedding of the query
         :param filters: Optional filters.
         :param top_k: How many documents to return.
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:212]: can we please document the `DocumentStoreError` which is raised in line 247?

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py [hunk 7]
```diff
@@ -211,7 +254,7 @@ def mongo_doc_to_haystack_doc(self, mongo_doc: Dict[str, Any]) -> Document:
         Converts the dictionary coming out of MongoDB into a Haystack document
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:254]: I'd make `mongo_doc_to_haystack_doc` private as it's not used outside this module

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py [hunk 8]
```diff
@@ -168,6 +210,7 @@ def embedding_retrieval(
         :param query_embedding: Embedding of the query
         :param filters: Optional filters.
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:211]: Should we document that filters are currently not implemented / supported?

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py [hunk 9]
```diff
@@ -95,10 +137,10 @@ def filter_documents(self, filters: Optional[Dict[str, Any]] = None) -> List[Doc
         Returns the documents that match the filters provided.
 
         For a detailed specification of the filters,
-        refer to the [documentation](https://docs.haystack.deepset.ai/v2.0/docs/metadata-filtering).
+        refer to the Haystack [documentation](https://docs.haystack.deepset.ai/v2.0/docs/metadata-filtering).
 
         :param filters: The filters to apply. It returns only the documents that match the filters.
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:142]: Should we document that filters are currently not implemented / supported?

File: integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py [hunk 10]
```diff
@@ -70,18 +97,19 @@ def run(
         top_k: Optional[int] = None,
     ) -> Dict[str, List[Document]]:
         """
-        Retrieve documents from the MongoDBAtlasDocumentStore, based on their embeddings.
+        Retrieve documents from the MongoDBAtlasDocumentStore, based on the provided embedding similarity.
 
         :param query_embedding: Embedding of the query.
         :param filters: Filters applied to the retrieved Documents. Overrides the value specified at initialization.
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py:103]: Should we document that filters are currently not implemented / supported?

File: integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py [hunk 11]
```diff
@@ -70,18 +97,19 @@ def run(
...
         :param top_k: Maximum number of Documents to return. Overrides the value specified at initialization.
-        :return: List of Documents similar to `query_embedding`.
+        :returns: A dictionary with the following keys:
+            - `documents`: List of Documents most similar to the given query_embedding
```
Comment (wochinge) [inline integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py:106]:
```suggestion
            - `documents`: List of Documents most similar to the given `query_embedding`
```

**Review by wochinge — CHANGES_REQUESTED** [review]

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py (hunk 8 above)
Comment (anakin87) [reply integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:211]: The PR for filters will be ready very soon, so we can skip this, I think.

File: integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py (hunk 10 above)
Comment (wochinge) [reply integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py:104]: @anakin87 what do you think?

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py (hunk 9 above)
Comment (wochinge) [reply integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:144]: @anakin87

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py (hunk 8 above)
Comment (wochinge) [reply integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:214]: ah okay 👍🏻

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py (hunk 9 above)
Comment (wochinge) [reply integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:144]: all good. saw the comment below

File: integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py (hunk 10 above)
Comment (wochinge) [reply integrations/mongodb_atlas/src/haystack_integrations/components/retrievers/mongodb_atlas/embedding_retriever.py:104]: all good. saw the comment below

**Pull request merged**